### ISSUE TITLE
[Bug] `/api/binary` endpoint returns the same response body regardless of request params

### ISSUE THREAD

**Issue by grwthomps** [ysfchn/SamFetch#6]
Not sure if I'm missing something but it seems like the `/api/binary` endpoint returns the same response body regardless of the params sent with the request.

Request:
`/api/binary?region=XAA&model=SM-G986U1&firmware=G986U1UEU1DUB7%2FG986U1OYM1DUB7%2FG986U1UEU1DUB7%2FG986U1UEU1DUB7`

Response:
```
{
  "display_name": "SSP",
  "size": 5991219840,
  "filename": "SM-G986U1_1_20210416234737_q8wahqy67m_fac.zip.enc4",
  "path": "/neofus/910/",
  "version": "R (Android 11)",
  "encrypt_version": 4,
  "size_readable": "5.58 GB",
  "decrypt_key": "60813b9c940b3e286de84bff7baaf403"
}
```

The `20210416` date of the firmware is what tipped me off because I'm trying to download a firmware from 2-28. So I also tried a couple of other firmwares (all pulled from the `/api/list` endpoint) and they all returned the same response as above. I'm guessing it's just returning info for the latest firmware every time but I'm not sure.

**Comment by ysfchn** [comment]
I just tried to display raw data from Kies's servers, and looks like it got the device firmware code correctly as expected.

![image](https://user-images.githubusercontent.com/54748183/117723059-95e1d100-b1ea-11eb-8609-861a9ec73e40.png)

And when I look at the https://doc.samsungmobile.com/SM-G986U1/XAA/doc.html  I can clearly see the release date.

![image](https://user-images.githubusercontent.com/54748183/117723670-5a93d200-b1eb-11eb-891a-4840edd23506.png)

But I don't know the exact reason why it returns that response to you.

I will research more about this problem and let you know if it is related to Kies / Samsung's servers itself or SamFetch.

**Comment by grwthomps** [comment]
Hmm that is strange. I just tried again with some different firmwares and got the same response. I even tried using Postman to send the request since I was using the openapi web interface before but that didn't change anything. Let me know if there's anything I need to try or test. Thanks for your help!

**Comment by ysfchn** [comment]
Probably this is the reason:

https://github.com/zacharee/SamloaderKotlin/issues/10#issuecomment-827292128

**Comment by grwthomps** [comment]
Yes, I think you're right. I just tested another tool on Windows 10 (SamFirm_Reborn) and it did the same thing so it seems like it's a change on Samsung's end.

**Comment by zacharee** [comment]
@ysfchn if you find any way around this, let me know.

**Comment by Martyn911** [comment]
Are you affiliated with the site [SamFirm](https://samfirm.shop/) ?
